keps/sig-scheduling/5234-dra-resourceslice-mixins/README.md

```diff
@@ -0,0 +1,902 @@
+- [x] (R) KEP approvers have approved the KEP status as `implementable`
+- [x] (R) Design details are appropriately documented
+- [x] (R) Test plan is in place, giving consideration to SIG Architecture and SIG Testing input (including test refactors)
+  - [ ] e2e Tests for all Beta API Operations (endpoints)
+  - [ ] (R) Ensure GA e2e tests meet requirements for [Conformance Tests](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md) 
+  - [ ] (R) Minimum Two Week Window for GA e2e tests to prove flake free
+- [ ] (R) Graduation criteria is in place
+  - [ ] (R) [all GA Endpoints](https://github.com/kubernetes/community/pull/1806) must be hit by [Conformance Tests](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md) 
+- [x] (R) Production readiness review completed
+- [x] (R) Production readiness review approved
+- [x] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation—e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+<!--
+**Note:** This checklist is iterative and should be reviewed and updated every time this enhancement is being considered for a milestone.
+-->
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+With Dynamic Resource Allocation (DRA), DRA drivers publish information about
+the devices that they manage in ResourceSlices. This information is used by the
+scheduler when selecting devices for user requests in ResourceClaims.
+
+With this KEP, DRA drivers can define metadata in mixins separately from specific
+devices and include them in a device by reference. This reduces the duplication
+in ResourceSlices and allows for more compact device definitions. It also enables
+defining devices with more attributes, capacities and consumed counters than without
+mixins. Mixins can also be used in counter sets.
+
+## Motivation
+
+DRA requires that drivers publish all available devices on a node/cluster in
+`ResourceSlice` objects. There are scenarios where the number of devices
+can be pretty large and each device might have a relatively large amount
+of metadata associated with it, primarily in the form of attributes and
+capacity. This has a few consequences:
+
+* Several of the devices might have similar metadata, resulting in a lot of
+  duplication between the published devices.
+* The size of the data required to specify each device reduces the number of
+  devices that can be defined in a single ResourceSlice without hitting the
+  Kubernetes size limit.
```

Comment:
can we clarify which size limit we're up against (e.g., is it the payload of the `spec` data in a CRD?)

Reply:
Added language to clarify.